imbi/clients/opensearch.py

```diff
@@ -117,73 +129,21 @@ async def index_document(self,
             json.dumps(normalize(sanitize_keys(document)), indent=0))
         await self.redis.sadd(self.PENDING_KEY, f'{index}:{document_id}')
 
-    async def search_fields(self) -> typing.List[typing.Dict]:
-        fields = []
-        index_mappings = await self._build_mappings()
-        for key, defn in index_mappings.items():
-            try:
-                fields.append({
-                    'name': key,
-                    'type': defn['type'],
-                    'count': 0,
-                    'scripted': False,
-                    'searchable': True,
-                    'aggregatable': True,
-                    'readFromDocValues': True
-                })
-            except KeyError:
-                pass
-
-        async with aiopg.create_pool(self.postgres_url) as pool:
-            async with pool.acquire() as conn:
-                async with conn.cursor(
-                        cursor_factory=extras.RealDictCursor) as cursor:
-                    await cursor.execute(
-                        'SELECT name, data_type FROM v1.project_fact_types;')
-                    async for row in cursor:
-                        fields.append({
-                            'name': row['name'].lower().replace(' ', '_'),
-                            'type': mappings.FACT_DATA_TYPES[row['data_type']],
-                            'count': 0,
-                            'scripted': False,
-                            'searchable': True,
-                            'aggregatable': True,
-                            'readFromDocValues': True
-                        })
-        return fields
+    async def search(self, index: str, query: str, max_results: int = 1000) \
+            -> typing.Dict[str, typing.List[dict]]:
```

Comment:
NIT, you can save yourself some typing by using the [PEP-585](https://peps.python.org/pep-0585/) syntax (pun intended)
```python
    async def search(self, index: str, query: str, max_results: int = 1000) -> dict[str, list[dict]]:
```

If we intend to support Python 3.7, then you will want to add
```python
from __future__ import annotations
```